Accidentially asking for void.class bridge on void returning method produces assert

The following silly code:

```
@WithBridgeMethods(value=void.class, castRequired=true) // for binary compatibility
protected void setParent(Foo newParent) { ...
```

produces assertion:

```
Exception in thread "main" java.lang.AssertionError: Unexpected operand size: V
    at com.infradna.tool.bridge_method_injector.MethodInjector$Transformer$SyntheticMethod.inject(MethodInjector.java:253)
    at com.infradna.tool.bridge_method_injector.MethodInjector$Transformer.visitEnd(MethodInjector.java:326)
    at org.objectweb.asm.ClassReader.accept(ClassReader.java:673)
    at org.objectweb.asm.ClassReader.accept(ClassReader.java:506)
    at com.infradna.tool.bridge_method_injector.MethodInjector.handle(MethodInjector.java:98)
    at com.infradna.tool.bridge_method_injector.ProcessMojo.execute(ProcessMojo.java:65)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:132)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:120)
    at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:347)
    at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:154)
    at org.apache.maven.cli.MavenCli.execute(MavenCli.java:582)
    at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:214)
    at org.apache.maven.cli.MavenCli.main(MavenCli.java:158)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:483)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
```

Where it should produce some more meaningful error.

--- a/injector/src/main/java/com/infradna/tool/bridge_method_injector/MethodInjector.java
+++ b/injector/src/main/java/com/infradna/tool/bridge_method_injector/MethodInjector.java
@@ -231,6 +231,8 @@ public class MethodInjector {
                     if (returnType.equals(Type.VOID_TYPE) || returnType.getClassName().equals("java.lang.Void")) {
                         // bridge to void, which means disregard the return value from the original method
                         switch (originalReturnType.getSize()) {
+                        case 0:
+                            throw new IllegalArgumentException("Cannot bridge " + name + " from void to void; did you mean to use a different type?");
                         case 1:
                             mv.visitInsn(POP);
                             break;

